src/main/java/io/hops/metadata/ndb/dalimpl/hdfs/FileProvenanceClusterj.java

```diff
@@ -0,0 +1,249 @@
+    ArrayList<ProvenanceLogEntryDto> added = new ArrayList<>(logEntries.size());
+    try {
+      for (FileProvenanceEntry logEntry : logEntries) {
+        added.add(createPersistable(logEntry));
+      }
+      session.savePersistentAll(added);
+    } finally {
+      session.release(added);
+    }
+  }
+
+  @Override
+  public void add(FileProvenanceEntry logEntry) throws StorageException {
+    HopsSession session = connector.obtainSession();
+    ProvenanceLogEntryDto dto = null;
+    try {
+      dto = createPersistable(logEntry);
+      session.savePersistent(dto);
+    } finally {
+      session.release(dto);
+    }
+  }
+
+  private ProvenanceLogEntryDto createPersistable(FileProvenanceEntry logEntry) throws StorageException {
+    HopsSession session = connector.obtainSession();
+    ProvenanceLogEntryDto dto = session.newInstance(ProvenanceLogEntryDto.class);
+    dto.setInodeId(logEntry.getInodeId());
+    dto.setOperation(logEntry.getOperation());
+    dto.setLogicalTime(logEntry.getLogicalTime());
+    dto.setTimestamp(logEntry.getTimestamp());
+    dto.setAppId(logEntry.getAppId());
+    dto.setUserId(logEntry.getUserId());
+    dto.setPartitionId(logEntry.getPartitionId());
+    dto.setProjectId(logEntry.getProjectId());
+    dto.setDatasetId(logEntry.getDatasetId());
+    dto.setParentId(logEntry.getParentId());
+    dto.setInodeName(logEntry.getInodeName());
+    dto.setProjectName(logEntry.getProjectName());
+    dto.setDatasetName(logEntry.getDatasetName());
+    dto.setP1Name(logEntry.getP1Name());
+    dto.setP2Name(logEntry.getP2Name());
+    dto.setParentName(logEntry.getParentName());
+    dto.setUserName(logEntry.getUserName());
+    dto.setXAttrName(logEntry.getXattrName());
+    dto.setLogicalTimeBatch(logEntry.getLogicalTimeBatch());
+    dto.setTimestampBatch(logEntry.getTimestampBatch());
+    dto.setDsLogicalTime(logEntry.getDsLogicalTime());
+    return dto;
+  }
+
+  private Collection<FileProvenanceEntry> createCollection(Collection<ProvenanceLogEntryDto> collection) {
+    ArrayList<FileProvenanceEntry> list = new ArrayList<>(collection.size());
+    for (ProvenanceLogEntryDto dto : collection) {
+      list.add(createProvenanceLogEntry(dto));
+    }
+    return list;
+  }
+
+  private FileProvenanceEntry createProvenanceLogEntry(ProvenanceLogEntryDto dto) {
+    return new FileProvenanceEntry(
+      dto.getInodeId(),
+      FileProvenanceEntry.Operation.valueOf(dto.getOperation()),
+      dto.getLogicalTime(),
+      dto.getTimestamp(),
+      dto.getAppId(),
+      dto.getUserId(),
+      dto.getPartitionId(),
+      dto.getProjectId(),
+      dto.getDatasetId(),
+      dto.getParentId(),
+      dto.getInodeName(),
+      dto.getProjectName(),
+      dto.getDatasetName(),
+      dto.getP1Name(),
+      dto.getP2Name(),
+      dto.getParentName(),
+      dto.getUserName(),
+      dto.getXAttrName(),
+      dto.getLogicalTimeBatch(),
+      dto.getLogicalTimeBatch(),
+      dto.getDsLogicalTime());
+  }
```

Comment:
Is there a plane to use these functions in the future? Write now they are not usefull as we only write to the database and never read from it.

Reply:
fixed - removed the methods